aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -296,4 +313,34 @@ void gpu_kernel_impl(TensorIterator& iter, const func_t& f) {
   }
 }
 
+template <typename func_t>
+void gpu_kernel_multiple_outputs_impl(TensorIterator& iter, const func_t& f) {
+  using traits = function_traits<func_t>;
+  using output_t = typename traits::result_type;
+  TORCH_INTERNAL_ASSERT(memory::detail::is_tuple<output_t>::value);
+  constexpr int num_outputs = std::tuple_size<output_t>::value;
+  TORCH_INTERNAL_ASSERT(num_outputs > 1);
+  constexpr int num_inputs = traits::arity;
+  constexpr int ntensors = num_outputs + num_inputs;
+
+  TORCH_INTERNAL_ASSERT(iter.can_use_32bit_indexing());
+  TORCH_INTERNAL_ASSERT(iter.ntensors() == ntensors);
+
+  at::detail::Array<char*, ntensors> data;
+  for (int i = 0; i < ntensors; i++) {
+    data[i] = (char*)iter.data_ptr(i);
+  }
+
+  int64_t numel = iter.numel();
+
+  at::detail::Array<ScalarType, ntensors> dtypes;
+  for (int i = 0; i < ntensors; i++) {
+    dtypes[i] = iter.tensor(i).scalar_type();
+  }
+
+  auto input_calc = make_input_offset_calculator<num_inputs, num_outputs>(iter);
+  auto output_calc = ::make_offset_calculator<num_outputs>(iter);
```

Comment:
I think you will need to use `make_output_offset_calculator` otherwise you will be using bytes as offset.